### PR TITLE
Refactor/switch to reportmi frontend

### PR DESCRIPTION
**Pull request by mec**
Report MI Frontend is a set of style components for the service:

https://github.com/Crown-Commercial-Service/reportmi-frontend

the admin application is already using the components and the supplier application should as well.

This PR:

- sets the version of GOVUK Frontend we load to 2.8.0
- changes the Rails config to use the GOVUK Frontend assets in development and production
- replaces the styles in the supplier app with those from the Report MI frontend, most of this is the same with minor changes, mainly to do with following the GOVUK Frontend code style which Report MI Frontend uses.
- with all that in place a couple of bits of markup needed new classes added
- we can now load the favicons with the Rails helper and it will work, no more blank assets
- removes the old GOVUK elements gems

I am not 100% sure on the changes in the Gemfile.lock, but I know this requires gem sass v3.7.3 so I had to update at least that, appreciate someones help on that!

### CODE REVIEW

File: app/views/layouts/application.html.haml [hunk 1]
```diff
@@ -4,11 +4,13 @@
     %meta{ content: 'text/html; charset=UTF-8', 'http-equiv': 'Content-Type' }/
     %title
       = yield(:page_title) || 'Report management information to CCS'
-
     %meta{ name: 'viewport', content: 'width=device-width, initial-scale=1' }/
     %meta{ 'http-equiv': 'x-ua-compatible', content: 'ie-edge' }/
     %meta{ name: 'format-detection', content: 'telephone=no' }/
 
+    = favicon_link_tag('favicon.ico')
```
Comment (tekin) [inline app/views/layouts/application.html.haml]: These govuk-frontend assets get compiled with their folder-structure intact, so this line will need to reference them with the full nesting, for it to work in production where live asset compilation is turned off, e.g.

```suggestion
    = favicon_link_tag('govuk-frontend/assets/images/favicon.ico')
```

Note: you can test this in development mode by disabling the asset-pipeline with `config.assets.compile = false` in development.rb:

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
+    = favicon_link_tag('govuk-apple-touch-icon.png', rel: 'apple-touch-icon', type: 'image/png')
```
Comment (tekin) [inline app/views/layouts/application.html.haml]: As above, use the full path to the asset:

```suggestion
    = favicon_link_tag('govuk-frontend/assets/images/govuk-apple-touch-icon.png', rel: 'apple-touch-icon', type: 'image/png')
```

File: app/views/layouts/errors.html.haml [hunk 3]
```diff
@@ -8,6 +8,9 @@
     %meta{ 'http-equiv': 'x-ua-compatible', content: 'ie-edge' }/
     %meta{ name: 'format-detection', content: 'telephone=no' }/
 
+    = favicon_link_tag(source='favicon.ico')
```
Comment (tekin) [inline app/views/layouts/errors.html.haml]: same as `application.html.haml`

File: app/views/layouts/application.html.haml (hunk 2 above)
Comment (mec) [reply app/views/layouts/application.html.haml]: @tekin how did you check this with `config.assets.compile = false` I then run into css and js issues?

Comment (tekin) [reply app/views/layouts/application.html.haml]: 1. `bundle exec rake assets:compile`
2. set `config.assets.compile = false`
3. (re)start the rails server

Comment (tekin) [reply app/views/layouts/application.html.haml]: Oh, you may also need to set `config.assets.debug = false`!

**Pull request merged**